### DISCUSSION TITLE
Federation as separate projects?

### DISCUSSION THREAD

**Discussion by omar-dulaimi** [MichalLytek/type-graphql#1110] [Q&A] [answered]
Hello,

Is it possible to divide a large schema to many smaller parts(projects) and connect them using federation?

I'm thinking of this to avoid running into issues caused by big schemas when generating their types, inputs, etc.

**Comment by MichalLytek** [comment, marked as answer]
https://github.com/MichalLytek/type-graphql/tree/master/examples/apollo-federation

**Reply by omar-dulaimi** [reply]
@MichalLytek I have seen that before. But I asked because I'm not sure if it's possible to divide the schema into multiple projects(not the same project with a port for each schema). I'm trying to make the integration with prisma generate smaller and more manageable number of files since each schema will live in a different project(different repo even).

**Reply by MichalLytek** [reply]
Have you tried that before asking the question? 🤔 It's "the same project with a port for each schema" to simulate multiple ports and multiple machines. They do not share decorated types nor resolvers.

**Reply by omar-dulaimi** [reply]
I honestly have not, and I appreciate your patience and time. I'll give it a try and see.
Thank you.

**Reply by omar-dulaimi** [reply]
This is exactly what I need. Thanks!
